gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -194,78 +179,15 @@ static __always_inline void count_v6(struct sock *sk, __u16 dport)
 		__atomic_add_fetch(val, 1, __ATOMIC_RELAXED);
 }
 
-static __always_inline void trace_v4(struct pt_regs *ctx, __u64 pid_tgid,
-				     struct sock *sk, __u16 dport,
-				     __u64 mntns_id, int ret)
-{
-	struct event *event;
-
-	event = gadget_reserve_buf(&events, sizeof(*event));
-	if (!event)
-		return;
-
-	__u64 uid_gid = bpf_get_current_uid_gid();
-
-	event->pid = pid_tgid >> 32;
-	event->tid = (u32)pid_tgid;
-	event->uid = (u32)uid_gid;
-	event->gid = (u32)(uid_gid >> 32);
-	event->src.version = event->dst.version = 4;
-	event->src.proto_raw = event->dst.proto_raw = IPPROTO_TCP;
-	BPF_CORE_READ_INTO(&event->src.addr_raw.v4, sk,
-			   __sk_common.skc_rcv_saddr);
-	BPF_CORE_READ_INTO(&event->dst.addr_raw.v4, sk, __sk_common.skc_daddr);
-	event->dst.port = dport;
-	event->src.port = BPF_CORE_READ(sk, __sk_common.skc_num);
-	event->mntns_id = mntns_id;
-	bpf_get_current_comm(event->comm, sizeof(event->comm));
-	event->timestamp_raw = bpf_ktime_get_boot_ns();
-	event->error_raw = -ret;
-
-	gadget_submit_buf(ctx, &events, event, sizeof(*event));
-}
-
-static __always_inline void trace_v6(struct pt_regs *ctx, __u64 pid_tgid,
-				     struct sock *sk, __u16 dport,
-				     __u64 mntns_id, int ret)
-{
-	struct event *event;
-
-	event = gadget_reserve_buf(&events, sizeof(*event));
-	if (!event)
-		return;
-
-	__u64 uid_gid = bpf_get_current_uid_gid();
-
-	event->pid = pid_tgid >> 32;
-	event->tid = (u32)pid_tgid;
-	event->uid = (u32)uid_gid;
-	event->gid = (u32)(uid_gid >> 32);
-	event->mntns_id = mntns_id;
-	event->src.version = event->dst.version = 6;
-	event->src.proto_raw = event->dst.proto_raw = IPPROTO_TCP;
-	BPF_CORE_READ_INTO(&event->src.addr_raw.v6, sk,
-			   __sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
-	BPF_CORE_READ_INTO(&event->dst.addr_raw.v6, sk,
-			   __sk_common.skc_v6_daddr.in6_u.u6_addr32);
-	event->dst.port = dport;
-	event->src.port = BPF_CORE_READ(sk, __sk_common.skc_num);
-	bpf_get_current_comm(event->comm, sizeof(event->comm));
-	event->timestamp_raw = bpf_ktime_get_boot_ns();
-	event->error_raw = -ret;
-
-	gadget_submit_buf(ctx, &events, event, sizeof(*event));
-}
-
 static __always_inline int exit_tcp_connect(struct pt_regs *ctx, int ret)
 {
 	__u64 pid_tgid = bpf_get_current_pid_tgid();
 	__u32 tid = pid_tgid;
 	struct sock **skpp;
 	struct sock *sk;
-	u64 mntns_id;
```

Comment:
This will conflict with #3517. I guess you will have to resolve the conflicts.

Reply:
I'll update this once that's merged.